### PR TITLE
feat: get authenticated user bb-14

### PR DESCRIPTION
**Pull request by Mikoyzskie**
- After refreshing the page, current user is received and saved in app state
- Use the token saved in localStorage to get current user and save that user to app state

### CODE REVIEW

File: apps/frontend/src/modules/users/slices/users.slice.ts [hunk 1]
```diff
@@ -4,15 +4,17 @@ import { DataStatus } from "~/libs/enums/enums.js";
 import { type ValueOf } from "~/libs/types/types.js";
 import { type UserGetAllItemResponseDto } from "~/modules/users/users.js";
 
-import { loadAll } from "./actions.js";
+import { getUser, loadAll } from "./actions.js";
 
 type State = {
 	dataStatus: ValueOf<typeof DataStatus>;
+	user: undefined | UserGetAllItemResponseDto;
```
Comment (fshabanov) [inline apps/frontend/src/modules/users/slices/users.slice.ts:11]:
```suggestion
	user: null | UserGetAllItemResponseDto;
```

File: apps/frontend/src/modules/users/slices/users.slice.ts [hunk 2]
```diff
@@ -4,15 +4,17 @@ import { DataStatus } from "~/libs/enums/enums.js";
...
 	users: UserGetAllItemResponseDto[];
 };
 
 const initialState: State = {
 	dataStatus: DataStatus.IDLE,
+	user: undefined,
```
Comment (fshabanov) [inline apps/frontend/src/modules/users/slices/users.slice.ts:17]:
```suggestion
	user: null,
```

File: apps/frontend/src/modules/users/slices/users.slice.ts [hunk 3]
```diff
@@ -28,6 +30,9 @@ const { actions, name, reducer } = createSlice({
 		builder.addCase(loadAll.rejected, (state) => {
 			state.dataStatus = DataStatus.REJECTED;
 		});
+		builder.addCase(getUser.fulfilled, (state, action) => {
+			state.user = { email: action.payload.email, id: action.payload.id };
```
Comment (fshabanov) [inline apps/frontend/src/modules/users/slices/users.slice.ts:34]:
```suggestion
			state.user = action.payload;
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by WilhelmWesser — APPROVED** [review]

**Review by revenets — APPROVED** [review]

File: apps/frontend/src/modules/users/slices/actions.ts [hunk 4]
```diff
@@ -15,4 +18,14 @@ const loadAll = createAsyncThunk<
 	return userApi.getAll();
 });
 
-export { loadAll };
+const getUser = createAsyncThunk<
```
Comment (fshabanov) [inline apps/frontend/src/modules/users/slices/actions.ts:21]:
```suggestion
const getAuthenticatedUser = createAsyncThunk<
```
Change everywhere

Comment (fshabanov) [inline apps/frontend/src/modules/users/slices/actions.ts:21]: You should call this action in ~/libs/components/app/app.tsx

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/modules/users/slices/actions.ts [hunk 5]
```diff
@@ -15,4 +18,14 @@ const loadAll = createAsyncThunk<
 	return userApi.getAll();
 });
 
-export { loadAll };
+const getAuthenticatedUser = createAsyncThunk<
+	UserGetAllItemResponseDto,
+	undefined,
+	AsyncThunkConfig
+>(`${sliceName}/get_user`, (_, { extra }) => {
```
Comment (fshabanov) [inline apps/frontend/src/modules/users/slices/actions.ts:25]:
```suggestion
>(`${sliceName}/get-authenticated-user`, (_, { extra }) => {
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**